### PR TITLE
[ZEPPELIN-5970][ZEPPELIN-5971] bugs occur when zeppelin.livy.tableWithUTFCharacters is enabled

### PR DESCRIPTION
**Pull request by levi0090**
### What is this PR for?
Fixed 2 bugs occur when **zeppelin.livy.tableWithUTFCharacters** is set to **True**. ref. commit ID  0fc1d7ba2081ebeed5c45d87c27ca57727f54099, 

The errors occur when:
1. **tableWithUTFCharacters** is set to True and INSERT/CREATE statement is executed.
2. **tableWithUTFCharacters** is set to True and SELECT query of a table that has non-primitive data (e.g. arrays).

### What type of PR is it?
 [MINOR] Bug Fix - JIRA-ticket ZEPPELIN-5970
 [MINOR] Bug Fix - JIRA-ticket ZEPPELIN-5971

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5970
* https://issues.apache.org/jira/browse/ZEPPELIN-5971

### How should this be tested?

1.  Enable **tableWithUTFCharacters** feature from livy interpreter.
3. Create non-select SQL statement, such as: CREATE table statement , you will receive an error, however, the table will be created !

```
%livy.scala
spark.sql("CREATE TABLE  test ( id int, name String, destination []String)")
```
 
4. INSERT a record with an array of values, will raise an exception as arrays are non-primitive data
```
%livy.scala
spark.sql("INSERT INTO test VALUES ('6', 'test', '[\'1'\,\'2\']')")
```

### Screenshots (if appropriate)
N/A

### Questions:
* Does the license files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by jongyoul** [comment]
@levi0090 Thank you for your contribution. I checked the code quickly and looks reasonable. BTW, could you please change the title of the PR including a ticket or the prefix [MINOR]?

File: livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java [hunk 1]
```diff
@@ -180,26 +181,50 @@ protected List<String> parseSQLJsonOutput(String output) {
...
             .replace("\n", "\\n")
             .replace("\t", "\\t"));
       }
       rows.add(StringUtils.join(cells, "\t"));
+
     }
     return rows;
   }
 
+  private Map<String, String> serialize(String jsonString) {
```
Comment (pan3793) [inline livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java:210]: maybe deserialize?

**Comment by pan3793** [comment]
@levi0090 I'm not sure the rules of Apache Zeppelin community, but generally, the bug fix requires a JIRA ticket

**Comment by pan3793** [comment]
@levi0090 would you mind providing the error stacktrace/message of the case you listed before applying this patch in the PR description? Such messages could help users who meet the same issue to identify the root cause.

File: livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java (hunk 1 above)
Comment (levi0090) [reply livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java]: Thanks for catching that. . Fixed. https://github.com/apache/zeppelin/pull/4675/commits/fbf6c98aa0ff91059bfd1d7f5fba0360a0a04bab

**Comment by levi0090** [comment]
I have created two Jira issues that describe both issues, for your reference.

**Comment by jongyoul** [comment]
@levi0090 Thank you for making tickets. Could you please add your ticket numbers in your PR title?

File: livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java [hunk 2]
```diff
@@ -180,26 +181,50 @@ protected List<String> parseSQLJsonOutput(String output) {
...
     }
     return rows;
   }
 
+  private Map<String, String> deserialize(String jsonString) {
+    Map<String, String> map = new HashMap<>();
+    Gson gson = new Gson();
+    JsonElement jsonElement = gson.fromJson(jsonString, JsonElement.class);
+    JsonObject jsonObject = jsonElement.getAsJsonObject();
+    for (String key : jsonObject.keySet()) {
```
Comment (Reamer) [inline livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java:215]: Please iterate over the entrySet.
See https://rules.sonarsource.com/java/RSPEC-2864/

**Review by Reamer — CHANGES_REQUESTED** [review]
Only one performance issue found. The rest looks good.

File: livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java (hunk 2 above)
Comment (levi0090) [reply livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java]: Done, thanks

**Comment by jongyoul** [comment]
@levi0090 I've added ticket prefixes. Could you please edit the title summarizing your contribution?

**Review by Reamer — DISMISSED** [review]

**Review by pan3793 — APPROVED** [review]

File: livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java [hunk 3]
```diff
@@ -180,26 +181,52 @@ protected List<String> parseSQLJsonOutput(String output) {
...
+      Map<String, String> retMap = deserialize(rowsOutput[i]);
+
       cells = new ArrayList<>();
       for (String s : header) {
         cells.add(retMap.getOrDefault(s, "null")
             .replace("\n", "\\n")
             .replace("\t", "\\t"));
       }
       rows.add(StringUtils.join(cells, "\t"));
+
```
Comment (pan3793) [inline livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java:205]: nit: avoid unnecessary change.

**Review by jongyoul — APPROVED** [review]

File: livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java (hunk 3 above)
Comment (Reamer) [reply livy/src/main/java/org/apache/zeppelin/livy/LivySparkSQLInterpreter.java]: removed

**Review by Reamer — APPROVED** [review]

**Comment by Reamer** [comment]
I will merge the changes with the master on 20.11 if no further comments are received.
